### DISCUSSION TITLE
DX enhancements for client directives

### DISCUSSION THREAD

**Discussion by przeprogramowani** [withastro/roadmap#816] [Proposal]
### Body

# Summary

I'd like to quickly discover missing client directives on components that are rendered, but frozen.

# Background & Motivation

From time to time I forget about client directives, spending time on debugging components that are visible, yet frozen. Especially when demoing something, experimenting with new frameworks within Astro or just being in hurry, this kills my productivity.

# Goals

There are at least two way of solving that:
1) Disallow rendering client components when they are used without any client directives
2) In dev mode, when developers are trying to interact with client components, let them know in some way that client directives are missing so interactivity is disabled

In my opinion, reducing negative sentiment is a goal worth pursuing so I'd appreciate hearing your thoughts on this. While I agree it's not a P1 feature to push during next major, it would be quite a nice DX improvements for people who are not that experienced with Astro.

What do we think about this problem?

**Comment by florian-lefebvre** [comment]
I think that could make a great dev toolbar app

**Comment by Princesseuh** [comment]
It's something we're hoping to somewhat mitigate with the x-ray mode of the dev toolbar.

In addition to the current feature showing the client directive used for a component, we'd like to add more relevant information in the same vein, including possible audits.

**Reply by przeprogramowani** [reply]
Would it be "opt-in" x-ray, or there would be a chance to get some proactive details about misconfigured directives as early as possible? (in terms of development journey)

**Reply by Princesseuh** [reply]
No, we'd probably add this to the audit part, so you'd get a red dot on it like you get for accessibility issues.
